packages/flutter_tools/lib/src/project_validator.dart

```diff
@@ -11,7 +13,116 @@ abstract class ProjectValidator {
+    GeneralInfoProjectValidator(),
   ];
 }
+
+class GeneralInfoProjectValidator extends ProjectValidator{
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async {
+    final YamlMap pubContent = loadYaml(project.pubspecFile.readAsStringSync()) as YamlMap;
+    final ProjectValidatorResult appNameValidatorResult = getAppNameResult(pubContent);
+    final String supportedPlatforms = getSupportedPlatforms(project);
+    if (supportedPlatforms.isEmpty) {
+      return <ProjectValidatorResult>[appNameValidatorResult];
+    }
+    final ProjectValidatorResult supportedPlatformsResult = ProjectValidatorResult(
+        name: 'Supported Platforms',
+        value: supportedPlatforms,
+        status: StatusProjectValidator.success
+    );
+    final ProjectValidatorResult isFlutterPackage = isFlutterPackageValidatorResult(pubContent);
+    final List<ProjectValidatorResult> result = <ProjectValidatorResult>[
+      appNameValidatorResult,
+      supportedPlatformsResult,
+      isFlutterPackage,
+    ];
+    if (isFlutterPackage.value == 'yes') {
+      final YamlMap flutterNode = pubContent['flutter'] as YamlMap;
+      result.add(materialDesignResult(flutterNode));
+      result.add(pluginValidatorResult(flutterNode));
+    }
+    return result;
+  }
+
+  ProjectValidatorResult getAppNameResult(YamlMap pubContent) {
+    final String? appName = pubContent['name'] as String?;
+    const String name = 'App Name';
+    if (appName == null) {
+      return const ProjectValidatorResult(
+          name: name,
+          value: 'name not found',
+          status: StatusProjectValidator.error
+      );
+    }
+    return ProjectValidatorResult(
+        name: name,
+        value: appName,
+        status: StatusProjectValidator.success
+    );
+  }
+
+  ProjectValidatorResult isFlutterPackageValidatorResult(YamlMap pubContent) {
+    String value;
+    StatusProjectValidator status;
+    if (pubContent.containsKey('flutter')) {
+      value = 'yes';
+      status = StatusProjectValidator.success;
+    } else {
+      value = 'no';
+      status = StatusProjectValidator.warning;
+    }
+
+    return ProjectValidatorResult(
+        name: 'Is Flutter Package',
+        value: value,
+        status: status
+    );
+  }
+
+  ProjectValidatorResult materialDesignResult(YamlMap flutterNode) {
+    bool isMaterialDesign;
+
+    if (flutterNode.containsKey('uses-material-design')) {
+      isMaterialDesign = flutterNode['uses-material-design'] as bool;
+    } else {
+      isMaterialDesign = false;
+    }
+
+    final String value = isMaterialDesign? 'yes' : 'no';
+    return ProjectValidatorResult(
+      name: 'Uses Material Design',
+      value: value,
+      status: StatusProjectValidator.success
+    );
+  }
+
+  String getSupportedPlatforms(FlutterProject project) {
+    final List<SupportedPlatform> supportedPlatforms = project.getSupportedPlatforms();
+    final List<String> allPlatforms = <String>[];
```

Comment:
This can be you'll prob :

```suggestion
    final List<String> allPlatforms = project.getSupportedPlatforms().map((SupportedPlatform platform) => platform.name);
```